includes/wikia/ExternalUser_Wikia.php

```diff
@@ -250,26 +250,35 @@ protected function addToDatabase( User &$User, $password, $email, $realname ) {
 		$dbw = wfGetDB( DB_MASTER, [], $wgExternalSharedDB );
 
 		try {
-			$dbw->insert(
-				'`user`',
-				[
-					'user_id' => null,
-					'user_name' => $User->mName,
-					'user_password' => $User->mPassword,
-					'user_newpassword' => $User->mNewpassword,
-					'user_newpass_time' => $dbw->timestamp( $User->mNewpassTime ),
-					'user_email' => $email,
-					'user_email_authenticated' => $dbw->timestampOrNull( $User->mEmailAuthenticated ),
-					'user_real_name' => $realname,
-					'user_options' => '',
-					'user_token' => $User->mToken,
-					'user_registration' => $dbw->timestamp( $User->mRegistration ),
-					'user_editcount' => 0,
-					'user_birthdate' => $User->mBirthDate
-				],
-				__METHOD__
-			);
-			$User->mId = $dbw->insertId();
+            $userId = null;
+            $result = null;
+            wfRunHooks( 'ExternalUserWikiaAddToDatabase', [ &$result, &$userId, $User, $password, $email, $realname ] );
+
+            if ( is_null( $result ) ) {
+                $dbw->insert(
+                    '`user`',
+                    [
+                        'user_id' => null,
+                        'user_name' => $User->mName,
+                        'user_password' => $User->mPassword,
+                        'user_email' => $email,
+                        'user_options' => '',
+                        'user_registration' => $dbw->timestamp($User->mRegistration),
+                        'user_editcount' => 0,
+                        'user_birthdate' => $User->mBirthDate
```

Comment:
We need to add "user_newpassword" as it is NOT NULL and doesn't have the DEFAULT value.